starfyre/component.py

```diff
@@ -19,6 +19,9 @@ class Component:
     js: str = ""
     # on any property change, rebuild the tree
 
+    is_custom: bool = False
+    inner_content: list = field(default_factory=list)
```

Comment:
I don't understand what are the responsibilities of these attributes. Can you please explain them again here?

Reply:
Sure.
I'm using `is_custom` property to check if we are now processing child nodes of our custom component. Such child nodes will go to a new list `inner_content` to separate them from "regular" children nodes (which are defined in the pyml-definition of the component).

`inner_content` list stores child nodes, which we then insert instead of <slot> tag.

Reply:
@SuelenKarbivnychyy 👋 

> I'm using is_custom property to check if we are now processing child nodes of our custom component. Such child nodes will go to a new list inner_content to separate them from "regular" children nodes (which are defined in the pyml-definition of the component).

By custom component do you mean `slot`?

Reply:
> By custom component do you mean `slot`?

Sorry, I do not mean that.

Let me rephrase it.

Our html nodes are represented by instances of `component` class. I defined a new list `inner_content` to hold children nodes of our custom components, for example:

<parent>
   <span>Hello World</span>
</parent>

- span will go to `innner_content` list because `<parent>` is our custom component.

is_custom == True for root tag of our custom components (e.g. for <span> in definition of parent.fyre).
Later, we use it to find “parent” tag for our `<slot>` element, and replace `<slot>` by whatever we have in `inner_content` list of `is_custom` component.


Please let me know if you need more clarification on it. :)

Reply:
Also, we have a list called `children`, which serves the same purpose as `inner_content` , why are we not making use of that??

Reply:
> Hello World
> * span will go to `innner_content` list because `<parent>` is our custom component.

For some reason, part of my comment is not showing up here. But when I click edit, I see how it should be. Weird, I will post my example again.

Reply:
> And instead of having a property like `is_custom`, we should be having a property like `is_slot` since slot is special component. And then the behaviours related to slot should be rendered accordingly.

`<slot>` is not a special component, `<slot>` is being a place hold for let us know where the actual `custom componente will go when transpiled it`.
I have created a test example that I will drop it here.

Reply:
> Also, we have a list called `children`, which serves the same purpose as `inner_content` , why are we not making use of that??

The `children` list is used only for component definition. 

We need to handle two different modes:
Component definition and component usage.

We use `children` list to handle children from component definition (in our test-application folder the component definition would be `parent.fyre`), and the `inner_content` list to handle the children aka inner content (where we use the custom component `<parent>`) on our app in `__init__.fyre`

Reply:
@SuelenKarbivnychyy looks like this conversation will benefit from a call. Does Saturday 730 pm bat work for you?

Reply:
@SuelenKarbivnychyy , 

> <slot> is not a special component, <slot> is being a place hold for let us know where the actual custom componente will go when transpiled it.

`<slot>` should be treated like a special component. As `<slot>` is a component according to our language grammar, just behaves in a different way.

Reply:
Also, this is not a good design practice. 

We are doing multiple things in the component class and hence introducing two different cases ourselves.

> We need to handle two different modes: Component definition and component usage. We use children list to handle children from component definition (in our test-application folder the component definition would be parent.fyre), and the inner_content list to handle the children aka inner content (where we use the custom component <parent>) on our app in __init__.fyre

Reply:
> The children list is used only for component definition. We need to handle two different modes: Component definition and component usage.

The children list is not being used for just definition. It is actually storing all the child components. And by `usage`, I think you mean `rendering`. We have different methods for `rendering`.